pkgs/top-level/haskell-packages.nix

```diff
@@ -414,8 +427,13 @@ in
       ghc912 = compiler.ghc9122;
       ghcHEAD = callPackage ../development/compilers/ghc/head.nix {
         bootPkgs =
-          # No suitable bindist packaged yet
-          bb.packages.ghc9101;
+          # No armv7l bindists are available.
+          if stdenv.buildPlatform.isAarch32 then
+            bb.packages.ghc984
+          else if stdenv.buildPlatform.isPower64 && stdenv.buildPlatform.isLittleEndian then
+            bb.packages.ghc984
+          else
```

Comment:
Should probably be checked whether we also need to avoid the bindist on Darwin like with e.g. 9101.

Reply:
Doesn't appear to be an issue (tested x86_64-darwin).